test/e2e/storage/sanity.go

```diff
@@ -995,3 +998,41 @@ func WaitForPodsWithLabelRunningReady(c clientset.Interface, ns string, label la
 		})
 	return pods, err
 }
+
+// Register list of volumes before test, using out-of-band host commands (i.e. not CSI API).
+// TODO: remove debug prints before merging
+func GetHostVolumes(numWorkers int) map[string][]string {
+	var cmd string
+	var hdr string
+	//var result map[string][]string
+	switch os.Getenv("TEST_DEVICEMODE") {
+	case "lvm":
+		// lvs adds many space (0x20) chars at end, we could squeeze
+		// repetitions using tr here, but TrimSpace() below strips those away
+		cmd = "sudo lvs --foreign --noheadings"
+		hdr = "LVM Volumes"
+	case "direct":
+		// ndctl produces multiline block. We want one line per namespace.
+		// Remove double quotes, delete lines dev:xyz and blockdev:xyz as these elems may change after reboot,
+		// remove newlines, then insert one at the end, clean some more.
+		cmd = "sudo ndctl list |tr -d '\"' |grep -v '^    dev:' |grep	-v '^    blockdev:' |tr -d '\n' |tr ']' '\n' |tr -d '[{}' |tr -d ' '"
+		hdr = "Namespaces"
+	}
+	result := make(map[string][]string)
+	for worker := 1; worker <= numWorkers; worker++ {
```

Comment:
Instead of taking a `numWorkers` argument, this function can iterate over all `_work/<cluster>/ssh.*` files. Then the caller doesn't have to figure out how many nodes there are in the cluster.

Reply:
I was having the same line of thinking. This approach may fail if the running VM set is not same as set of ssh.N files, but then something is wrong or system was manually altered, i.e. edge cases which are already out of ordinary, so I think let's try this.